examples/script/concolic.py

```diff
@@ -164,15 +163,16 @@ def symbolic_run_get_cons(trace):
     m2.verbosity(VERBOSITY)
     m2.register_plugin(f)
 
-    @m2.hook(main_end)
-    def x(s):
-        with m2.locked_context() as ctx:
+    def on_gen_testcase(mcore, state, name, msg):
+        with mcore.locked_context() as ctx:
             readdata = []
-            for name, fd, data in s.platform.syscall_trace:
+            for name, fd, data in state.platform.syscall_trace:
                 if name in ('_receive', '_read') and fd == 0:
                     readdata.append(data)
             ctx['readdata'] = readdata
-            ctx['constraints'] = list(s.constraints.constraints)
+            ctx['constraints'] = list(state.constraints.constraints)
+
+    m2.subscribe('will_generate_testcase', on_gen_testcase)
```

Comment:
is will_generate_testcase the best event? maybe some kind of terminate_state event would be better?